pkg/kube/util/boshdns/dns.go

```diff
@@ -283,12 +283,24 @@ func (dns *boshDomainNameService) createCorefile(namespace string) (string, erro
 				if !found {
 					continue
 				}
-				for i := 0; i < instanceGroup.Instances; i++ {
-					id := fmt.Sprintf("%s-%d", target.InstanceGroup, i)
-					from := strings.Replace(alias.Domain, "_", id, 1)
-					serviceName := instanceGroup.IndexedServiceName(dns.ManifestName, i)
-					to := fmt.Sprintf("%s.%s.svc.%s", serviceName, namespace, clusterDomain)
-					rewrites = append(rewrites, dnsTemplate(from, to, target.Query))
+				if len(instanceGroup.AZs) > 0 {
+					for azIndex := range instanceGroup.AZs {
+						for i := 0; i < instanceGroup.Instances; i++ {
```

Comment:
This is too deeply nested for my taste: for,for,if,if,for,for. Maybe some of that code can go into a separate function?

There is also some overlap with the else case below.

Reply:
True, Thanks for the comment. Hope, I fixed it now